libs/sdk-ui-dashboard/src/model/commandHandlers/scheduledEmail/initializeAutomationsHandler.ts

```
@@ -108,22 +108,29 @@ export function* initializeAutomationsHandler(
 
         // Set filters according to provided automationId
         if (automationId) {
+            const targetAutomation = automations.find((a) => a.id === automationId);
+            const targetWidget = targetAutomation?.metadata?.widget;
+
+            if (!targetAutomation || !targetWidget) {
+                return;
```

Maybe throw an error here instead?

---

I removed the return,  because the saga was obviously not returning and I needed it as mentioned in my previous comment 😅